Added EXT_texture_mirror_clamp_to_edge extension

Groovounet opened the pull request:
OpenGL ES is missing an extension to expose MIRROR_CLAMP_TO_EDGE texture wrap mode.

TobiasHector commented:
Is anyone actually exposing this, or planning on doing so? I don't see it on http://opengles.gpuinfo.org/gles_extensions.php. As a general rule of thumb, I don't think we should take extensions into the ES registry without at least one implementer signing up to do it. I mean we can, I'm just not sure what the point would be other than to confuse the ecosystem by implying something is supported when it's not?

I guess I'll put this on the next WG telecon agenda, and we'll see if anyone wants to sign up/comment.

Groovounet commented:
I am actually just hoping to get vendor onboard with this. Considering that https://vulkan.gpuinfo.org/listreports.php?extension=VK_KHR_sampler_mirror_clamp_to_edge is largely supported, I have good hopes.

ianromanick commented:
I'd be in favor of removing the ES 3.0 requirement.  I believe we have some GPUs that support GL_ARB_mirror_clamp_to_edge that only do ES 2.0.  That means an interaction section would need to be added that says, "If OpenGL ES 3.0 or GL_OES_texture3D is not supported, remove references to TEXTURE_WRAP_R."

Maybe this could replace the copy-and-paste issue about compatibility profile. :)

Groovounet commented:
Hi Ian,

I updated the extension proposal to require OpenGL ES 2.0 instead of OpenGL ES 3.0 and removed the copy pasted issue.

Thanks,
Christophe

dgkoch commented on extensions/EXT/EXT_texture_mirror_clamp_to_edge.txt:
```diff
@@ -0,0 +1,151 @@
+    OpenGL ES 2.0 is required.
+
+    This extension is written against the OpenGL ES 3.2 Specification.
+
```
This extension interacts with OES_texture_3D. 

This extension interacts with OpenGL ES 3.0.

This extension interacts with OpenGL ES 3.2.

dgkoch commented on extensions/EXT/EXT_texture_mirror_clamp_to_edge.txt:
```diff
@@ -0,0 +1,151 @@
+    TEXTURE_WRAP_T, or TEXTURE_WRAP_R parameter is set to REPEAT,
+    MIRRORED_REPEAT, or MIRROR_CLAMP_TO_EDGE.
+
+Dependencies on EXT_texture3D or equivalent
```
OES_texture_3D (EXT_texture3D is an GL extension, not ES).

dgkoch commented on extensions/EXT/EXT_texture_mirror_clamp_to_edge.txt:
```diff
@@ -0,0 +1,151 @@
+    The error INVALID_ENUM is generated when TexParameter* is called
+    with a target of TEXTURE_RECTANGLE and the TEXTURE_WRAP_S,
+    TEXTURE_WRAP_T, or TEXTURE_WRAP_R parameter is set to REPEAT,
+    MIRRORED_REPEAT, or MIRROR_CLAMP_TO_EDGE.
```
This error is not applicable for ES (no TEXTURE_RECTANGLE target).

dgkoch commented on extensions/EXT/EXT_texture_mirror_clamp_to_edge.txt:
```diff
@@ -0,0 +1,151 @@
+
+    If OpenGL ES 3.0 or equivalent is not supported, then ignore all
+    references to sampler objects and SamplerParameter* functions.
+
```
Add: 
Dependencies on OpenGL ES 3.2 or equivalent

If OpenGL ES 3.2 or equivalent is not supported, then ignore all references to the TexParameterI* and SamplerParameterI* functions.

dgkoch commented on extensions/Makefile:
```diff
@@ -17,7 +17,7 @@
 # python3 is required. Sorry, python2 dudes.
 
-PYTHON = python3
+PYTHON = python
```
I don't think this change should be included.

dgkoch commented on extensions/EXT/EXT_texture_mirror_clamp_to_edge.txt:
```diff
@@ -0,0 +1,157 @@
+
+Status
+
+    Draft on September 7, 2017.
```
Did you mean to switch this to Draft?

Groovounet replied:
I mean it but fixed. 
Probably I should have post it for review as draft :D

dgkoch commented:
Looks good otherwise (and including the header changes).

Groovounet commented:
Erm, when I do "make" in the "extensions" directory it tells me "nothing to be done for 'all'". So the PHP files are not generated. Am I missing something?

oddhack commented:
@Groovounet it looks to me like the 'extensions' directory is up to date, which is why nothing happens - the generated PHP file is already current with registry.py, and part of the PR diff. Maybe you did it earlier and forgot?

dgkoch commented:
LGTM now. @oddhack please go ahead and merge this now.

I think the headers will still need to be re-generated after this is accepted
